### PR TITLE
clickhouse validation: use temporary table

### PR DESCRIPTION
**Pull request by serprex**
avoids leaking table due to errors

### CODE REVIEW

File: flow/connectors/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -75,44 +75,34 @@ func (c *ClickHouseConnector) ValidateCheck(ctx context.Context) error {
 	}
 	validateDummyTableName := "peerdb_validation_" + shared.RandomString(4)
 	// create a table
-	err := c.exec(ctx, fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
-		id UInt64
-	) ENGINE = ReplacingMergeTree ORDER BY id;`,
-		validateDummyTableName+"_temp"))
-	if err != nil {
+	if err := c.exec(ctx,
+		fmt.Sprintf(`CREATE TEMPORARY TABLE IF NOT EXISTS %s (id UInt64) ENGINE = ReplacingMergeTree ORDER BY id;`, validateDummyTableName),
```
Comment (iskakaushik) [inline flow/connectors/clickhouse/clickhouse.go:79]: given that the intent is to validate the grants, can we confirm that create temp table and create table have the same grants?

Comment (serprex) [reply flow/connectors/clickhouse/clickhouse.go:79]: Looking over, `CREATE TEMPORARY TABLE` & `CREATE TABLE` are separate

https://clickhouse.com/docs/en/sql-reference/statements/show#show-grants

Probably best to use `SHOW GRANTS` to minimize pollution. But have to recursively run on roles granted to user

Comment (heavycrystal) [reply flow/connectors/clickhouse/clickhouse.go:79]: should we combine a normal table with a goroutine based drop
after addressing the race condition

Comment (serprex) [reply flow/connectors/clickhouse/clickhouse.go:79]: We don't need a goroutine, validation is not slow

**Pull request closed without merging**